suite/tests/pthreads/ptsig.c

```diff
@@ -146,6 +156,11 @@ main(int argc, char **argv)
         exit(1);
     }
 
+    /* We may get only one SIGUSR1 if kernels merges. */
+    while (signal_received == 0)
+        thread_yield();
```

Comment:
Cleaner to use a condition variable and let the kernel wake us up only when it's set, instead of yielding.

Reply:
Done.